### PR TITLE
Fixes Saving a Project in a new tab

### LINKED ISSUE #1358: Saving Online should open in new tab
**Describe the bug**
Whenever a user clicks to save his project [online], he gets redirected to the Edit Project Screen losing his work area. 

If he wants to work further on his project he needs to click on update project button from the Edit Project Screen then click on Launch Simulator to get his circuit back.

**Expected behavior**
The Edit Project Screen should open in a new page / After clicking on the Update Project button he should be redirected to the simulator screen back.

### PR DESCRIPTION
**Pull request by Shivansh2407**
Fixes #1358 

#### Describe the changes you have made in this PR -
I have fixed Saving the project in a new tab. Now it opens in a new tab. Also the loader has been removed because on opening a new page the loader will be of no use in the simulator .

### Screenshots of the changes (If any) -
Patch43

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by Shivansh2407** [comment]
@nitin10s and @satu0king Please review this 👍 . I will be fixing the codeclimate issues soon.

**Comment by Shivansh2407** [comment]
> Don't remove the loadingIcon from the save function just for guest users, saving online has the same function logged in or not, saving a project online needs that.
> 
> edit: the loading icon is required when you're already logged in, & save your changes online.

I have removed the loading icon because when the user is logged in then on saving the project a new tab is opened and hence we do not require the loading icon anywhere. Also if I include it then when we save the project, it opens in new tab and the loading icon is displayed in the simulator tab which I think will not look good. For logged out users, the save will work similarly but after logging in. I would like to know the views of @nitin10s and @satu0king on this.

**Comment by abstrekt** [comment]
If project is already saved don't open a new tab for that,

**Comment by abstrekt** [comment]
just move the loading.fadein inside the else statement.

File: public/js/data.js [hunk 1]
```diff
@@ -436,21 +436,19 @@ function generateImageForOnline() {
 // Function to save data online
 function save() {
     projectSaved = true;
-
-    $('.loadingIcon').fadeIn();
```
Comment (abstrekt) [inline public/js/data.js:440]: Move this inside the else statement, not remove it.

Comment (Shivansh2407) [reply public/js/data.js:440]: Fixed in my recent commit 👍

**Comment by Shivansh2407** [comment]
@nitin10s and @satu0king please review 👍

File: public/js/data.js (hunk 1 above)
Comment (abstrekt) [reply public/js/data.js:440]: you're welcome.

**Comment by Shivansh2407** [comment]
@nitin10s please review 👍

**Comment by Shivansh2407** [comment]
@nitin10s @satu0king @tachyons please review this 👍

**Comment by nitin10s** [comment]
Basically: For the first time, it should create an instance, after that if an instance exists, rather than making a new instance, it should update the project already saved, with the new changes.

**Comment by Shivansh2407** [comment]
I will fix this soon @nitin10s

**Comment by shubhankarsharma00** [comment]
@Shivansh2407 are you still working on this? This one's a bit tricky.

**Comment by satu0king** [comment]
> @Shivansh2407 are you still working on this? This one's a bit tricky.

I agree. You will need to create a new instance and get back to project ID and update the state (url + other metadata), this will almost definitely need some changes in the backend as well.

**Comment by satu0king** [comment]
Closing PR as this is old UI. Thanks!

**Pull request closed without merging**